webapp/player/index.html

```diff
@@ -0,0 +1,215 @@
+                    videoSrcElement.setAttribute('src', videoSrc);
+                    videoPlayer.load();
+                    videoPlayer.play();
+                  }
+
+                  break;
+
+                case "trp":
+                  
+                  // create the Div
+                  var terminalDiv = document.createElement("div")
+                  document.body.appendChild(terminalDiv)
+
+                  let trpSrc = `${windowURL.protocol}//${windowURL.host}/jet/jrec/pull/${sessionId}/${recordingInfo.files[0].fileName}?token=${token}`
+
+                  loadFile(trpSrc, function(trpFileContent) {
+  
+                    var castFileContent = convertTRPtoCast(trpFileContent);
+
+                    // make the file a base64 embedded src url
+                    var url = "data:text/plain;base64,"+btoa(castFileContent);
+                    var player = new XtermPlayer.XtermPlayer(url,terminalDiv);  
+
+                    // need a slight delay to play waiting for it to load
+                    setTimeout(function() {
+                      player.play();
+                    }, 500);
+                  });
+
+                  break;
+                case "cast":
+
+                  // create the Div
+                  var terminalDiv = document.createElement("div")
+                  document.body.appendChild(terminalDiv)
+
+                  let castSrc = `${windowURL.protocol}//${windowURL.host}/jet/jrec/pull/${sessionId}/${recordingInfo.files[currentIndex].fileName}?token=${token}`
+
+                  loadFile(castSrc, function(castFileContent) {
+
+                    // make the file a base64 embedded src url
+                    var url = "data:text/plain;base64,"+btoa(castFileContent);
+                    var player = new XtermPlayer.XtermPlayer(url,terminalDiv);  
+
+                    // need a slight delay to play waiting for it to load
+                    setTimeout(function() {
+                      player.play();
+                    }, 500);
+                  });
+                  
+                  break;
+              }
+
+            } else {
+                console.error('Request failed. Returned status of ' + request.status);
+            }
+          }
+        }
+
+        request.open('GET', videoSrcInfo, true);
+        request.send();
+
+        function loadFile(fileName, onLoad) {
+          const req = new XMLHttpRequest();
+          req.open("GET", fileName, true);
+          req.responseType = "arraybuffer";
+          req.onload = (event) => {
+            const arrayBuffer = req.response;
+            if (arrayBuffer) {
+              const byteArray = new Uint8Array(arrayBuffer);
+              onLoad(byteArray);
+            }
+          };
+          req.send(null);
+        }
+
+        function convertTRPtoCast(fileArray) {
```

Comment:
we have our answer, Paul baked in a live TRP to .cast converter directly in the Javascript of the player, so it'll be able to play the older TRP file format as well :)